core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -685,6 +835,14 @@ private[spark] class ExecutorAllocationManager(
       stageAttemptToNumRunningTask.values.sum
     }
 
+    def totalRunningTasksPerResourceProfile(rp: Int): Int = {
+      val attempts = resourceProfileIdToStageAttempt.getOrElse(rp, Set.empty).toSeq
+      // attempts is a Set, change to Seq so we keep all values
+      attempts.map { attempt =>
+        stageAttemptToNumRunningTask.getOrElseUpdate(attempt, 0)
```

Comment:
Hi, @tgravescs here do we need to update `0` back to stageAttemptToNumRunningTask instead of using:

```scala
stageAttemptToNumRunningTask.getOrElseUpdate(attempt, 0)
```
If we update this without `allocationManager.synchronized`, it could have issues.